code/game/turfs/simulated.dm

```diff
@@ -55,13 +60,15 @@
 			else
 				wet_overlay = image('icons/effects/water.dmi', src, "wet_static")
 		overlays += wet_overlay
-	if(!infinite)
-		spawn(rand(790, 820)) // Purely so for visual effect
-			if(!istype(src, /turf/simulated)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
-				return
-			MakeDry(wet_setting)
+	if(time == INFINITY)
+		return
+	if(time == null)
+		time =	rand(790, 820)
+	addtimer(CALLBACK(src, /turf/simulated./proc/MakeDry, wet_setting), time)
 
 /turf/simulated/proc/MakeDry(wet_setting = TURF_WET_WATER)
+	if(!istype(src, /turf/simulated)) //Because turfs don't get deleted, they change, adapt, transform, evolve and deform. they are one and they are all.
+		return
```

Comment:
I know these lines were already here, but do you know why this is needed? If the callback for this proc was called on a `turf/simulated`, why do we need to check that `src` is not that type anymore? And the comment doesn't really help :smile:

Reply:
Yeah no clue why it's there. It probably does nothing currently. Will test it without and see the results if I go change a turf

Reply:
It did nothing yet caused a runtime if it happened. The turf was changed and it tries to call the proc.
But if it's say space it will still runtime. Fixed in the last commit